code/datums/components/proximity_monitor.dm

```diff
@@ -1,139 +1,472 @@
  */
 /datum/component/proximity_monitor
-	var/atom/owner
-	/// A list of currently created `/obj/effect/abstract/proximity_checker`s in use with this component.
+	can_transfer = TRUE
+	var/name = "Proximity detection field"
+	/// The primary atom the component is attached to and that will be receiving `HasProximity()` calls. Usually the `parent`.
+	var/atom/hasprox_receiver
+	/**
+	 * The atom that the component is listening to for movement. Null by default, but this can change.
+	 * For example, if the receiver gets put into a humans's pocket, the host will become the human.
+	 */
+	var/atom/host
+	/// The radius of the field, in tiles.
+	var/radius
+	/// A list of currently created [/obj/effect/abstract/proximity_checker] in use with this component.
 	var/list/proximity_checkers
+	/// The type of checker object that should be used for the main field.
+	var/field_checker_type = /obj/effect/abstract/proximity_checker
+	/// Should the parent always detect proximity and update the field on movement, even if it's not on a turf?
+	var/always_active
 
-/datum/component/proximity_monitor/Initialize()
+/datum/component/proximity_monitor/Initialize(_radius = 1, _always_active = FALSE)
 	. = ..()
-	if(!isatom(parent))
+	if(!ismovable(parent) && !isturf(parent)) // No areas or datums allowed.
 		return COMPONENT_INCOMPATIBLE
-	owner = parent
+	ASSERT(_radius >= 1)
+	hasprox_receiver = parent
+	radius = _radius
+	always_active = _always_active
 	create_prox_checkers()
 
+	if(isturf(hasprox_receiver.loc))
+		toggle_checkers(TRUE)
+		return
+	else if(always_active)
+		toggle_checkers(TRUE)
+	// We only want a host to track if we're `always_active`.
+	host = get_atom_on_turf(host)
+
 /datum/component/proximity_monitor/Destroy(force, silent)
+	hasprox_receiver = null
+	host = null
 	QDEL_LIST(proximity_checkers)
-	owner = null
 	return ..()
 
 /datum/component/proximity_monitor/RegisterWithParent()
 	. = ..()
-	if(ismovable(parent))
-		RegisterSignal(parent, COMSIG_MOVABLE_MOVED, .proc/HandleMove)
+	// Override vars are set to TRUE here to allow for component transfering.
+	if(ismovable(hasprox_receiver))
+		RegisterSignal(hasprox_receiver, COMSIG_MOVABLE_MOVED, .proc/handle_move, TRUE)
+	if(host && ismovable(host))
+		RegisterSignal(host, COMSIG_MOVABLE_MOVED, .proc/handle_move, TRUE)
 
 /datum/component/proximity_monitor/UnregisterFromParent()
 	. = ..()
-	if(ismovable(parent))
-		UnregisterSignal(parent, COMSIG_MOVABLE_MOVED)
+	if(ismovable(hasprox_receiver))
+		UnregisterSignal(hasprox_receiver, COMSIG_MOVABLE_MOVED)
+	if(host && ismovable(host))
+		UnregisterSignal(host, COMSIG_MOVABLE_MOVED)
+
+/datum/component/proximity_monitor/PostTransfer()
+	if(!ismovable(parent) && !isturf(parent))
+		return COMPONENT_INCOMPATIBLE
+	host = null
+	hasprox_receiver = parent
+	if(!isturf(hasprox_receiver.loc))
+		host = get_atom_on_turf(host)
```

Comment:
```suggestion
		host = get_atom_on_turf(hasprox_receiver)
```
`host` here is `null` and I believe you meant `hasprox_receiver` here. That or `hasprox_receiver.loc`

Reply:
Yeah should be `hasprox_receiver`, good catch.